File: app/src/main/java/Controller/ChatsRecyclerViewAdapter.java
package Controller;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.QuickContactBadge;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import com.whiteside.insta.ui.ChatActivity;
import com.whiteside.insta.R;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.EventListener;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreException;

import java.util.List;

import Model.Profile;

import static android.content.ContentValues.TAG;

public class ChatsRecyclerViewAdapter extends RecyclerView.Adapter<ChatsRecyclerViewAdapter.ViewHolder> {

    Context context;
    List<String> friends;
    FirebaseFirestore fStore;

    public ChatsRecyclerViewAdapter(final Context context, List<String> friends) {
        this.context = context;
        this.friends = friends;

        fStore = FirebaseFirestore.getInstance();
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.chat_card_view, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, final int position) {
        holder.ll.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, ChatActivity.class);
                intent.putExtra("UID", friends.get(position));
                context.startActivity(intent);
            }
        });

        String UID = friends.get(position);
        fStore.collection("Users")
                .document(UID)
                .addSnapshotListener(new EventListener<DocumentSnapshot>() {
                    @Override
                    public void onEvent(@Nullable DocumentSnapshot ds, @Nullable FirebaseFirestoreException e) {
                        if (e != null) {
                            Log.e(TAG, e.getStackTrace().toString());
                        } else if (ds.exists()) {
                            Profile profile = ds.toObject(Profile.class);

                            Bitmap bitmap = BitmapFactory.decodeByteArray(profile.getProfilePic().toBytes()
                                    , 0, profile.getProfilePic().toBytes().length);
                            holder.badge.setImageBitmap(bitmap);
                            holder.name.setText(profile.getName());
                        }
                    }
                });
    }


    @Override
    public int getItemCount() {
        return friends.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        QuickContactBadge badge;
        TextView name;
        LinearLayout ll;

        public ViewHolder(@NonNull View v) {
            super(v);

            badge = v.findViewById(R.id.profile_image);
            name = v.findViewById(R.id.name);
            ll = v.findViewById(R.id.relativeLayout);
        }
    }
}

File: app/src/main/java/com/whiteside/insta/adapter/ChatsRecyclerViewAdapter.kt
package com.whiteside.insta.adapter

import com.whiteside.insta.model.Profile
import android.content.ContentValues
import android.content.Context
import android.content.Intent
import android.graphics.BitmapFactory
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.QuickContactBadge
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.firestore.FirebaseFirestore
import com.whiteside.insta.R
import com.whiteside.insta.ui.chat.ChatActivity

class ChatsRecyclerViewAdapter(var context: Context?, var friends: List<String>) : RecyclerView.Adapter<ChatsRecyclerViewAdapter.ViewHolder>() {
    var fStore: FirebaseFirestore
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.card_view_chat, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.ll.setOnClickListener {
            val intent = Intent(context, ChatActivity::class.java)
            intent.putExtra("UID", friends[position])
            context!!.startActivity(intent)
        }
        val UID = friends[position]
        fStore.collection("Users")
                .document(UID)
                .addSnapshotListener { ds, e ->
                    if (e != null) {
                        Log.e(ContentValues.TAG, e.stackTrace.toString())
                    } else if (ds!!.exists()) {
                        val profile = ds.toObject(Profile::class.java)
                        val bitmap = BitmapFactory.decodeByteArray(profile!!.profilePic!!.toBytes(), 0, profile.profilePic!!.toBytes().size)
                        holder.badge.setImageBitmap(bitmap)
                        holder.name.text = profile.name
                    }
                }
    }

    override fun getItemCount(): Int {
        return friends.size
    }

    inner class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var badge: QuickContactBadge
        var name: TextView
        var ll: LinearLayout

        init {
            badge = v.findViewById(R.id.profile_image)
            name = v.findViewById(R.id.name)
            ll = v.findViewById(R.id.relativeLayout)
        }
    }

    init {
        fStore = FirebaseFirestore.getInstance()
    }
}